util/execdetails/execdetails.go

```diff
@@ -68,44 +71,33 @@ type RuntimeStat struct {
 	rows int64
 }
 
-// NewRuntimeStats creates new executor collector.
-func NewRuntimeStats() RuntimeStats {
-	return RuntimeStats(make(map[string]*RuntimeStat))
+// NewRuntimeStatsColl creates new executor collector.
+func NewRuntimeStatsColl() *RuntimeStatsColl {
+	return &RuntimeStatsColl{stats: make(map[string]*RuntimeStats)}
 }
 
-// GetRuntimeStat gets execStat for a executor.
-func (e RuntimeStats) GetRuntimeStat(planID string) *RuntimeStat {
-	if e == nil {
-		return nil
-	}
-	runtimecStat, exists := e[planID]
+// Get gets execStat for a executor.
+func (e *RuntimeStatsColl) Get(planID string) *RuntimeStats {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+	runtimeStats, exists := e.stats[planID]
 	if !exists {
-		runtimecStat = &RuntimeStat{}
-		e[planID] = runtimecStat
-	}
-	return runtimecStat
-}
-
-func (e RuntimeStats) String() string {
-	var buff bytes.Buffer
-	buff.WriteString("(")
-	for planID, stat := range e {
-		buff.WriteString(planID + ":" + stat.String() + ",")
+		runtimeStats = &RuntimeStats{}
+		e.stats[planID] = runtimeStats
 	}
-	buff.WriteString(")")
-	return buff.String()
+	return runtimeStats
 }
 
 // Record records executor's execution.
-func (e *RuntimeStat) Record(d time.Duration, rowNum int) {
+func (e *RuntimeStats) Record(d time.Duration, rowNum int) {
 	atomic.AddInt32(&e.loop, 1)
 	atomic.AddInt64(&e.consume, int64(d))
 	atomic.AddInt64(&e.rows, int64(rowNum))
 }
 
-func (e *RuntimeStat) String() string {
+func (e *RuntimeStats) String() string {
 	if e == nil {
 		return ""
 	}
-	return fmt.Sprintf("time:%f, loops:%d, rows:%d", time.Duration(e.consume).Seconds()*1e3, e.loop, e.rows)
+	return fmt.Sprintf("time:%fms, loops:%d, rows:%d", time.Duration(e.consume).Seconds()*1e3, e.loop, e.rows)
```

Comment:
May be we can use `%v` to let the system libaray handle the time unit.

Reply:
IMHO, fixed `ms` is better.

Reply:
If it takes too long, we need to mentally parse it to something like `min` or `hour`?

Reply:
it's not normal has min or hour executue time  and more 000000 is ok for me too :rofl:

how do u think about ~? @zz-jason  @XuHuaiyu

Reply:
DONE~~~PTAL @lamxTyler  :laughing: